### ISSUE TITLE
`throw-new-error` false positive on class extension

### ISSUE THREAD

**Issue by Nicholaiii** [sindresorhus/eslint-plugin-unicorn#2654]
The rule is erroring in cases where it's not relevant, eg extending another class into a new error type :) 

unicorn/throw-new-error

```ts
//   3:33  error  Use `new` when creating an error  unicorn/throw-new-error
//------------------------------v
export class QueryError extends Data.TaggedError('QueryError')<{
  cause: unknown
}> {}
```

**Comment by kachkaev** [comment]
`Data.TaggedError` comes from https://github.com/Effect-TS/effect (11K+ stars & [growing rapidly](https://www.star-history.com/#Effect-TS/effect&Date)). Trying this library in a side project and am facing an incompatibility with unicorn.

Relevant docs: https://effect.website/docs/error-management/expected-errors/

Since we are dealing with a rather popular library, I believe that we need to add a special case to avoid this false positive.
